Source/WebKit/UIProcess/ios/WKContentViewInteraction.mm

```diff
@@ -12317,7 +12317,11 @@ - (void)continueContextMenuInteraction:(void(^)(UIContextMenuConfiguration *))co
 
 #if ENABLE(DATA_DETECTION)
     if ([(NSURL *)linkURL iTunesStoreURL]) {
-        [self continueContextMenuInteractionWithDataDetectors:continueWithContextMenuConfiguration];
+#if ENABLE(NETWORK_CONNECTION_INTEGRITY)
+        if (!_page->networkConnectionIntegrityPolicies().contains(WebCore::NetworkConnectionIntegrity::Enabled))
```

Comment:
Are we sure this is what we want to check? Instead of checking for an ephemeral session?